manifests/plugin/postgresql.pp

```diff
@@ -10,7 +10,10 @@
 
   include ::collectd
 
-  $_manage_package = pick($manage_package, $::collectd::manage_package)
+  $_manage_package    = pick($manage_package, $::collectd::manage_package)
+  $databases_defaults = { 'ensure' => $ensure }
+  $queries_defaults   = { 'ensure' => $ensure }
+  $writers_defaults   = { 'ensure' => $ensure }
```

Comment:
Why create all these identical hashes?

@bastelfreak Did you not like https://github.com/voxpupuli/puppet-collectd/pull/874#discussion_r267584362 ?

Reply:
looool i thought this too. It;'s one way of separating defaults for each resource

Reply:
```
    collectd::plugin::postgresql::writer { $resource:
      * => { 'ensure' => $ensure } + $attributes,
    }
```
better?

Reply:
With `create_resources` we had to pass a hash for the defaults (although it didn't strictly need to be assigned to a variable first).

Since we just have a single default property (`ensure`), putting it directly in the resource declaration is probably cleaner??

Reply:
> ```
>     collectd::plugin::postgresql::writer { $resource:
>       * => { 'ensure' => $ensure } + $attributes,
>     }
> ```
> better?

No need to merge hashes.  Just...
```puppet
collectd::plugin::postgresql::writer { $resource:
  ensure => $ensure,
  *      => $attributes,
}
```

Reply:
That depends on `$attributes` not containing an override for `ensure` though. (which might be a perfectly reasonable assumption?)

Reply:
they do contain them....
https://github.com/voxpupuli/puppet-collectd/blob/master/manifests/plugin/postgresql/writer.pp#L3

Reply:
Actually... I don't think that is a safe assumption.
` * => { 'ensure' => $ensure } + $attributes,` works for me.

Reply:
ok. well, we still get one thing... replacement of create_resources :)